pkgs/development/compilers/factor-lang/mk-factor-application.nix

```diff
@@ -0,0 +1,115 @@
+{ lib
+, writeText
+, makeWrapper
+, factor-lang
+, factor-no-gui
+, librsvg
+, gdk-pixbuf
+} @ initAttrs:
+
+{ name ? "${attrs.pname}-${attrs.version}"
+
+, factor-lang ? if enableUI then initAttrs.factor-lang else initAttrs.factor-no-gui
+
+, stdenv ? factor-lang.stdenv
```

Comment:
Could we remove this argument and use `mkFactorApplication.override { stdenv = ...; }` instead?

Currently, `stdenv.mkDerivation` is the only build helper that takes fixed-point arguments (finalAttrs: { ... }), and people are putting effort into bringing such support to other build helpers. Most implementation involves passing the received `finalAttrs: { ... }` down the base derivation (`stdenv.mkDerivation` in this case), and it's almost impossible to get the chosen `stdenv` out of the unfixed arguments without knowing what `stdenv.mkDerivation` to pass it into.

Reply:
I dropped it. This was residual from its early implementation.